src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -266,138 +343,181 @@ impl WorkunitStore {
     let started = Workunit {
       name,
       span_id: span_id.clone(),
-      parent_id: parent_id.clone(),
+      parent_id,
       state: WorkunitState::Started {
         start_time: std::time::SystemTime::now(),
       },
       metadata,
     };
-    let mut inner = self.inner.lock();
     if !self.rendering_dynamic_ui {
       started.log_workunit_state()
     }
+    let sender = self.heavy_hitters_data.started_workunits_tx.lock();
```

Comment:
Ditto narrowing the scope of the lock. In this case you might be able to do:
```
self.heavy_hitters_data.started_workunits_tx.lock().send(started.clone()).unwrap();
```
...to cause the lock to be released immediately? If not, a block.